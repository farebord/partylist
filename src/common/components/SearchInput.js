import React from 'react'
import Input from '@material-ui/core/Input'

const SearchInput = () => 
    <Input
        fullWidth 
        placeholder="Search..."
    />

export default SearchInput;