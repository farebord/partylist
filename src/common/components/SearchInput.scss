%flex-centered {
  display: flex;
  align-items: center;
  justify-content: center;
}

.fieldContainer {
  @extend %flex-centered;
  padding: 0 0.25rem;
  flex: 1 0 auto;
  margin: 0 0.625rem;
}

.searchInput {
  margin-left: 0.5rem;
  flex: 1 1 auto;
}

.inputIcon {
  @extend %flex-centered;
  padding: 0.5rem;
  margin: 0;
}