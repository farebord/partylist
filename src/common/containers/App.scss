$default-padding-width: 0.625rem;

%flex-centered {
  display: flex;
  align-items: center;
  justify-content: center;
}

%flex-grow-only {
  flex: 1 0 50%;
}

%background-image-centered {
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
}

.wrapper {
  display: flex;
  height: 100vh;
  flex-direction: row;
  background-color: #f5f5f5
}

.searchContainer, .controlContainer, .tracksContainer {
  @extend %flex-centered;
  flex-direction: row;
  width: 100%;
  padding: $default-padding-width 0;
}

.playerContainer {
  @extend %flex-centered;
  @extend %flex-grow-only;
  flex-direction: column;
}

.middleContainer {
  @extend %flex-centered;
  @extend %background-image-centered;
  flex: 1 1 auto;
  width: 100%;
  box-sizing: border-box;
  border-left: inset $default-padding-width transparent;
  border-right: inset $default-padding-width transparent;
  background-size: contain;
}

.imageContainer{
  @extend %flex-centered;
  @extend %flex-grow-only;
  @extend %background-image-centered;
}

.tracksContainer {
  justify-content: flex-start;
  align-items: flex-start;
  flex: 1 1 auto;
  overflow-y: scroll;
  padding: 0;
}

@media (min-width: 1025px) {

  .tracksContainer {
    display: flex;
  }

  .middleContainer {
    display: none;
  }  
}

@media (max-width: 1024px) {
 
  .imageContainer {
    display: none !important;
  }

  .tracksContainer {
    display: none;
  }

  .middleContainer {
    display: flex;
  }
   
}