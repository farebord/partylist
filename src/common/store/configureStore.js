import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const configureStore = () => {
  const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(thunk)),
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('../reducers', () => {
      // eslint-disable-next-line global-require
      const nextRootReducer = require('../reducers').default;
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
};

export default configureStore;
