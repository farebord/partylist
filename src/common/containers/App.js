import React from 'react'
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/styles';

import Grid from '@material-ui/core/Grid'
import SearchInput from '../components/SearchInput'

const styles = {
  containerGrid: {
    height: '100vh',
    padding: '10px'
  }
}

class App extends React.Component { 
  componentDidMount(){
    console.log(this.props)
  }
  render() {
    return (
      <Grid container className={this.props.classes.containerGrid}>
        <Grid item xs={12}>
          <SearchInput />
        </Grid>
        <Grid item xs={12}>

        </Grid>
        <Grid item xs={12}>

        </Grid>
      </Grid>
    )
  }
}


const mapStateToProps = () => ({
  
});

const mapDispatchToProps = dispatch => ({
  
})

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(App));
