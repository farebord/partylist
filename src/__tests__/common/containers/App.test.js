import React from 'react';
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'
import App from 'common/containers/App';
import SearchInput from 'common/components/SearchInput'

const mockStore = configureStore()
const store = mockStore()

describe('App component should', () => {
    it('render correctly', () => {
        const component = shallow(<App store={store} />)
        expect(component).toMatchSnapshot();
    });
    
})
