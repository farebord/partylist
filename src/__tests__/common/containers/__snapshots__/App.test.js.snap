// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(WithStyles(App))
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Naked": [Function],
      "displayName": "WithStyles(App)",
      "options": Object {},
      "propTypes": Object {
        "classes": [Function],
        "innerRef": [Function],
      },
      "render": [Function],
      "useStyles": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Naked": [Function],
        "displayName": "WithStyles(App)",
        "options": Object {},
        "propTypes": Object {
          "classes": [Function],
          "innerRef": [Function],
        },
        "render": [Function],
        "useStyles": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;

exports[`renders App correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(WithStyles(App))
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "Naked": [Function],
      "displayName": "WithStyles(App)",
      "options": Object {},
      "propTypes": Object {
        "classes": [Function],
        "innerRef": [Function],
      },
      "render": [Function],
      "useStyles": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "Naked": [Function],
        "displayName": "WithStyles(App)",
        "options": Object {},
        "propTypes": Object {
          "classes": [Function],
          "innerRef": [Function],
        },
        "render": [Function],
        "useStyles": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): Object {
    "storeSubscription": undefined,
  },
}
`;
